cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/similarity/JaccardSimilarity.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.similarity
+
+import com.twitter.cassovary.graph.{DirectedGraph, Node}
+import com.twitter.cassovary.graph.GraphDir.GraphDir
+
+class JaccardSimilarity(val graph: DirectedGraph[Node]) extends Similarity {
+
+  def calculateSimilarity(dir: GraphDir, u: Int, v: Int): Double = {
+    val neighborsOfU = getNeighbors(dir, u)
```

Comment:
It seems like both of these similarity algorithms require a lot of the same information which is leading to quite a bit of duplication (especially if we start adding more of these).  Maybe, and I'm not sure this is the best option, you could do something like this:

```scala
case class PairSimilarityMetrics(u: Int, v: Int) {
  val neighborsOfU = getNeighbors(dir, u)
  val neighborsOfV = getNeighbors(dir, v)
  lazy val commonNeighbors = neighborsOfU intersect neighborsOfV
  lazy val unionNeighbors = neighborsOfU union neighborsOfV
}
```

Then you could collect all of the metrics that you need in the case class?